pkg/liquidity-mining/test/ChildChainGaugeRegistry.test.ts

```diff
@@ -0,0 +1,135 @@
+  let admin: SignerWithAddress, other: SignerWithAddress;
+  let registry: Contract;
+
+  const factoryVersion = JSON.stringify({
+    name: 'ChildChainGaugeFactory',
+    version: '1',
+    deployment: 'test-deployment',
+  });
+  const productVersion = JSON.stringify({
+    name: 'ChildChainGauge',
+    version: '0',
+    deployment: 'test-deployment',
+  });
+
+  before('setup signers', async () => {
+    [, admin, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy gauge registry', async () => {
+    const vault = await Vault.create();
+    const balToken = await deploy('TestBalancerToken', { args: [admin.address, 'Balancer', 'BAL'] });
+
+    const mockVE = await deploy('v2-solidity-utils/TestToken', { args: ['Voting Escrow', 'veBAL', 18] });
+    const mockBPT = await deploy('v2-solidity-utils/TestToken', { args: ['Balancer Pool Test token', 'BPTST', 18] });
+
+    const veDelegationProxy = await deploy('VotingEscrowDelegationProxy', {
+      args: [vault.address, mockVE.address, ZERO_ADDRESS],
+    });
+
+    const pseudoMinter = await deploy('L2BalancerPseudoMinter', { args: [vault.address, balToken.address] });
+
+    gaugeImplementation = await deploy('ChildChainGauge', {
+      args: [veDelegationProxy.address, pseudoMinter.address, vault.authorizerAdaptor.address, productVersion],
+    });
+
+    gaugeFactory = await deploy('ChildChainGaugeFactory', {
+      args: [gaugeImplementation.address, factoryVersion, productVersion],
+    });
+
+    otherGaugeFactory = await deploy('ChildChainGaugeFactory', {
+      args: [gaugeImplementation.address, factoryVersion, productVersion],
+    });
+
+    registry = await deploy('ChildChainGaugeRegistry', {
+      args: [vault.address, pseudoMinter.address, gaugeFactory.address],
+    });
+
+    await vault.grantPermissionGlobally(await actionId(pseudoMinter, 'addGaugeFactory'), admin.address);
+    await vault.grantPermissionGlobally(await actionId(registry, 'addGauge'), admin.address);
+    await vault.grantPermissionGlobally(await actionId(registry, 'removeGauge'), admin.address);
+
+    await pseudoMinter.connect(admin).addGaugeFactory(gaugeFactory.address);
+
+    let tx = await gaugeFactory.create(mockBPT.address);
+    let event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+    gauge = await deployedAt('ChildChainGauge', event.args.gauge);
+
+    tx = await otherGaugeFactory.create(mockBPT.address);
+    event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+    otherGauge = await deployedAt('ChildChainGauge', event.args.gauge);
+  });
+
+  describe('addGauge', () => {
+    it('can add a valid gauge to the registry', async () => {
+      const tx = await registry.connect(admin).addGauge(gauge.address);
+
+      expectEvent.inReceipt(await tx.wait(), 'GaugeAdded', { gauge: gauge.address });
+      expect(await registry.totalGauges()).to.eq(1);
+      expect((await registry.getGauges(0, 1))[0]).to.eq(gauge.address);
+    });
```

Comment:
You could split this `it` block in 3 small tests:
- `it('emits an event')` --> checks only that the event is emitted in the tx.
- `it('increments total gauges') --> checks total gauges before and after calling `addGauge` (after = before + 1, you don't need to hardcode the size).
- `it('adds the gauge at the end') --> checks that the gauge can be retrieved with `getGauges` using the right indexes. You can call `totalGauges` before to know what index to use as start so that you don't need to hard code it.

Then, you can add all those 3 `it` blocks inside a function called `itAddsGaugeCorrectly` and use it below.

Reply:
Hardcoding values can be beneficial in that it eliminates reliance on potentially faulty functions. In fact, during testing, it is essential to ensure that specific inputs yield predetermined outputs. If your inputs are not fixed, you may be susceptible to elusive bugs.